sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2527,13 +2538,13 @@ def is_provided_ips_valid(ips: Optional[List[Optional[str]]]) -> bool:
                 worker_ip_max_attempts=max_attempts,
                 get_internal_ips=True)
 
-        assert len(cluster_external_ips) == len(cluster_internal_ips), (
+        assert len(cluster_feasible_ips) == len(cluster_internal_ips), (
             f'Cluster {self.cluster_name!r}:'
             f'Expected same number of internal IPs {cluster_internal_ips}'
-            f' and external IPs {cluster_external_ips}.')
+            f' and external IPs {cluster_feasible_ips}.')
 
         internal_external_ips: List[Tuple[str, str]] = list(
```

Comment:
Naming: should we go over vars/comments in this func and replace external -> feasible when it makes sense?

Reply:
I added a comment here, but keep the variable name the same. Reason: changing the variable name `self.stable_internal_external_ips` can cause a significant amount change in the code and can cause backward compatibility issue that we need to fix with the `__setstate__`. Also, the name `internal_feasible_ips` can be a bit confusing. I would prefer to keep the var name the same. wdyt?